core/utils/utils.go

```diff
@@ -17,12 +17,32 @@
 package utils
 
 import (
+	"bufio"
+	"context"
 	"encoding/binary"
+	"fmt"
 	"os"
+	"sync"
+	"time"
 
-	"github.com/satori/go.uuid"
+	uuid "github.com/satori/go.uuid"
 )
 
+type traceIDKey struct{}
+
+// TraceID returns traceid provided by WithTraceField and ContextWithTrace helpers.
+func TraceID(ctx context.Context) string {
```

Comment:
Are we really need duplicate trace id manipulation helpers from `instrumentation/inslogger` here?